lib/pluginmanager/package.rb

```diff
@@ -0,0 +1,61 @@
+require "logstash/util/compress"
+
+class LogStash::PluginManager::Package < LogStash::PluginManager::Command
+
+  option "--tgz", :flag, "compress package as a tar.gz file", :default => !LogStash::Environment.windows?
+  option "--zip", :flag, "compress package as a zip file", :default => LogStash::Environment.windows?
+  option "--clean", :flag, "clean up the generated dump of plugins", :default => false
+  option "--override", :flag, "Override a previously generated package file", :default => false
+
+  def execute
+    signal_error("File #{LogStash::Environment::GEMFILE_PATH} does not exist or is not writable, aborting") unless File.writable?(LogStash::Environment::GEMFILE_PATH)
+    puts("Packaging plugins for offline usage")
+    if override?
+      File.delete(target_file)
+    else
+      validate_target_file
+    end
```

Comment:
You could do something like this.


```ruby
def delete_target_file?
  return true if override?
  puts("File #{target_file} exist, do you want to override it to continue? (Y/N)")
  override = ( "y" == STDIN.gets.strip.downcase ? true : false)
end
```

def validate_target_file
  if file.exist?(target_file) && delete_target_file?
    File.delete(target_file)
  else
    signal_error("Package creation cancelled, a previously generated package exist at location: #{target_file}, move this file to safe place and run the command again")
   exit
end
end